Source/Server/MqttService.cpp

```diff
@@ -103,57 +84,48 @@ namespace MQTT {
 			
 			clientState->ConnectionFlags = package.VariableHeader.VariableLevel;
 
-			auto shouldDisconnectClient = !(RuleEngine({
-				{new ClientConnectedRule(packageClientId, m_ClientStates), false},
-				{new CorrectProtocolNameRule(protocolName), true},
-				{new Protocol311Rule(package.VariableHeader.Level), true},
-				{new ConnectReservedFlagSetRule(package.VariableHeader.VariableLevel), false},
-				{new IsCredentialFlagIncorrectRule(package.VariableHeader.VariableLevel), false},
-				{new ConnectWillRule(clientState, package.VariableHeader.VariableLevel, package.Payload.WillMessage), true}
-			}).Run());
+			auto action = Protocol::Validators::ConnectValidator()
+				.ValidateClient(package, m_ClientStates, clientState);
+
+			clientState->ConnectionIdentifier = client.GetIdentifier();
 
-			if (shouldDisconnectClient)
+			switch (action)
 			{
+			case MQTT::Protocol::Validators::ConnectValidator::Disconnect:
 				m_Server->Disconnect(client);
-				return;
+				break;
+			case MQTT::Protocol::Validators::ConnectValidator::RejectUserIdentifier:
+				m_Server->Send(client, m_Manager.GenerateConnectAckMessage(Protocol::Refused_Identifier_Rejected));
+				break;
+			case MQTT::Protocol::Validators::ConnectValidator::RejectProtocolLevel:
+				m_Server->Send(client, m_Manager.GenerateConnectAckMessage(Protocol::Refused_Unacceptable_Protocol_Version));
+				break;
+			case MQTT::Protocol::Validators::ConnectValidator::ContinueState:
+				m_Server->Send(client, m_Manager.GenerateConnectAckMessage(Protocol::Accepted));
+				clientState->IsConnected = true;
+				break;
+			case MQTT::Protocol::Validators::ConnectValidator::CreateNewState:
+				m_ClientStates.push_back(clientState);
+				m_Server->Send(client, m_Manager.GenerateConnectAckMessage(Protocol::Accepted));
+				clientState->IsConnected = true;
+				break;
 			}
+		}
 
-			std::vector<unsigned char> message;
-			auto shouldContinueSession = (RuleEngine({
-				{new ContinueSessionRule(package.VariableHeader.VariableLevel, packageClientId, m_ClientStates), true}
-			}).Run());
+		void MqttService::OnClientDisconnect(const Client& client, const Protocol::DisconnectPackage& package)
+		{
+			//TODO: Remove will message when storage of it is implemented.
+			DisconnectClientState(client);
+		}
 
-			if (shouldContinueSession)
+		void MqttService::DisconnectClientState(const Client& client)
+		{
+			for (auto& arr_client : m_ClientStates)
```

Comment:
arr_client -> arr_Client?

Reply:
Renamed to clientState: (clientState : m_ClientStates)